### ISSUE TITLE
Can we talk about naming?

### ISSUE THREAD

**Issue by Jaykul** [RamblingCookieMonster/PSDepend#80]
Why is it that the main install function is named `Invoke-PSDepend` instead of `Install-BuildDependencies` or something appropriate like that?

**Comment by gaelcolas** [comment]
It's not really installing IMO... but I feel like Restore would fit well.
Naming is hard.

**Comment by Jaykul** [comment]
I guess I'm pretty confused about who and what you would use this for apart from build-time. Especially when you say it's not really installing. 

What are you using it for?

**Comment by RamblingCookieMonster** [comment]
Borrowed a lot from PSDeploy, thus, Invoke-PSDeploy became Invoke-PSDepend (couldn't come up with a reason to change it back then).

A few things I've used it for:

* Dependencies for spot-solutions that might require packages, files from the web, repos, files from the file system, etc.
* Above... along with option to save these to a .requirements folder, and add that to path/psmodulepath, etc.
* Simplify instructions for co-workers - run invoke-psdepend, vs. you need XYZ various things from various sources, and the various questions that came with it
* When working on a module, a quick way to pull in dependencies for a project I haven't published yet (probably not efficient to keep in line with the RequiredModules, but, it works)

For the most part, yeah, the end result is an install, but `Invoke-PSDepend` has some other things built in:

* It has a super janky idea of letting dependency types handle `install`, `test`, and `import` as appropriate, each of which have a `Install-Dependency` esque command
* By default it will look for and invoke files it finds matching `'\.depend\.psd1$|.*requirements.psd1$'`
* Allows specifying arbitrary `InputObject` hashtables that equate to the content of a psd1
* Allows filtering for specific tags
* Allows specifying different actions, e.g. `-Install`, `-Test`, `-Import` 
* Allows overriding the Target for all dependencies (e.g. I might include a sane default in my dependency file, that others could override at runtime)

Basically... it's a convenient way to invoke psdepend : P

Cheers!

**Comment by rlvandaveer** [comment]
Invoke-PSDepend is very similar to how you execute other similar modules e.g., psake: Invoke-Psake, pester: Invoke-Pester, etc.

**Comment by Jaykul** [comment]
That doesn't make "invoke" a good verb. In fact, of these tools are _doing it wrong_, because they are all following Pester's **anti-pattern** approach to naming. Remember, Pester is the module that couldn't be bothered to follow verb-noun, and introduced Ruby-like syntax to PowerShell.

`Invoke-PSDepend` is only a good name if "Depend" is the name of a command or method.

Having said that, a year ago, I cared. Now, I don't. It's been badly named all this time, and nobody else cares (and I decided not to use it) so it doesn't matter.
